src/app/static-resources/static/js/updateFrontend.js

```diff
@@ -0,0 +1,67 @@
+addEventListener("DOMContentLoaded", (event) => {
+  updateContent();
+});
+
+
+/**
+ * Check if zaken have loaded, if not, try to load async
+ */
+function updateContent() {
+  const body = document.querySelector('body');
+  if(body.dataset?.loaded=='false') {
+    getData();
+  }
+}
+
+async function getData() {
+  const token = document.querySelector('meta[name=xsrf-token]').content;
+  const response = await fetch('', {
+    method: 'GET',
+    credentials: "same-origin",
+    headers: { 
+      'Accept': 'application/json',
+      'xsrftoken': token,
+    }
+  });
+
+  if(!response.ok) {
+    throw new Error('Network response was not OK');
+  };
+  const data = await response.json();
+  if(data.elements) {
+    for(el of data.elements) {
+      replaceElement(el);
+    }
+  } 
+  // if(data.error) {
+    // showErrorMessage(sendingButton, data.error);
+  // }
+};
+
+
+/**
+ * Transform an html-string to an actual nodetree
+ * 
+ * @param {string} html the valid HTML as as string
+ * @returns {Node} the html node
+ */
+function htmlStringToElement(html) {
```

Comment:
Wat als ik een formulier met HTML erin verstuur en deze wordt later getoond op mijn-nijmegen. Kan dat voor problemen zorgen?

Reply:
Zoals we de functie nu gebruiken nóg niet, omdat we deze code alleen aanroepen bij de eigen response, en daar geen user-content in zit (zaak-gegevens worden door ons gegenereerd). Maar, waar we in de templates HTML escapen, is dit wel een omweg dus risico. Als we bijv. op de zaak-detailpagina de content van het submission-object gaan tonen kan het fout gaan.

Wat we nu doen is volgens mij safe (schiet er vooral op!):
- In de JSON-response zit een 'html-string', die wordt gegenereerd door mustache. De mustache-template doet de normale escaping, dus als we een pagina genereren met onvertrouwde content zit die in de 'normale' mustache-variabele met {{}} er omheen, die escaped. Alleen de onveilige variant {{{}}} kan dan een risico zijn (net als nu). 
- Als we voor de json-response die we hier verwerken altijd dezelfde templates gebruiken als voor de 'backend'-rendered html, blijft de escaping op dezelfde manier werken, en blijven we veilig.

Reply:
Dat klinkt inderdaad logisch. Zolang mustache de HTML blijft generen die de 'webservice' terug geeft wordt de html escaped zou dit ook geen probleem moeten opleveren als we de html string weer omzetten in HTML hier.

Wat betreft de inhoud van de html, die is inderdaad door ons gegenereerd. Maar kan mij voorstellen dat we deze logica misschien ooit op een andere plek hergebruiken, dus is het goed om er alvast over na te denken.